rfcs/0000-multiple-kuadrant-instances.md

```diff
@@ -0,0 +1,152 @@
+
+# Motivation
+[motivation]: #motivation
+
+The main benefit of multiple Kuadrant instances in a single cluster is that it allows dedicated Kuadrant's services for tenants.
+
+Dedicated Kuadrant deployment brings lots of benefits. Just to name a few:
+* Protection against external traffic load spikes. Other tenant's traffic spikes does not affect Authorino/Limitador throughput and delay as it would when shared.
+* No need to have cluster administrator role to deploy a kuadrant instance. One tenant administrator can manage gateways, Limitador and Authorino instances (including deployment modes).
+* The cluster administrator gets control and visibility across all the Kuadrant instances, while the tenant administrator only gets control over their specific gateway(s), Limitador and Authorino instances.
+* (looking for ideas for more benefits)...
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+### Kuadrant instance definition
+![](https://i.imgur.com/BfOXfnB.png)
+
+A kuadrant is composed of:
+* One Limitador deployment instance
+* One Authorino deployment instance
+* A list of dedicated gateways.
+
+Some properties to highlight:
+
+* The policies are not included as part of the kuadrant instances.
+* The Kuadrant instance is not enclosed by k8s namespaces.
+* Gateways are not shared between kuadrant instances. Each gateway is managed by a single kuadrant instance.
+* The control plane has cluster scope and will be shared between instances. In other words, it is only in the data plane that each Kuadrant instance has dedicated services and resources.
+* Each kuadrant instance owns one instance (possibly multiple replicas, though) of Limitador and one instance of Authorino. Those instances are shared among all gateways included in the kuadrant instance.
+
+In the following diagram policies RLP 1 and KAP 1 are applied in the instance *A* and the policies RLP 2 and KAP 2 are applied in the instance *B*.
+
+![](https://i.imgur.com/yChVsT6.png)
+
+### All the gateways referenced by a single policy must belong to the same kuadrant instance
+
+The Gateway API allows, in its latest version  [v1beta1](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.CommonRouteSpec), an HTTPRoute to have multiple gateway parents. Thus, a kuadrant policy might technically target multiple gateways managed by multiple kuadrant instances. Kuadrant does **not** support this use case.
+
+![](https://i.imgur.com/ZpsBf4i.png)
+
+The main reason is related to the rate limiting capability. The limits specified in the RateLimit Policy would be enforced per kuadrant instance basis (provided by Limitador instance). Thus, traffic hitting one gateway would see different rate limiting counters than traffic hitting the other gateway. The user would expect X rps and actually it would be X rps per gateway. For consistency reasons, when this configuration happens, the control plane will reject the policy.
```

Comment:
While I think this is a good first step and we indeed should reject such policies initially, I think there might be a way to support this as we get to architect for multiple cluster support. The use-case isn't much different. What I'm thinking is that each limitador would know about each other. Upon hitting a "shared" limit, they'd start sharing the counters for it. Which, other than latency being lower and cross communication being easier, there isn't much differences between the two applications. 

Sharing this thought here as an FYI and sharing a possibly path forward. Again, it looks perfectly acceptable to have that limitation initially.